tests/unit/admin/test-class-wc-rest-payments-settings-controller.php

```diff
@@ -130,6 +130,28 @@ public function test_get_settings_returns_available_payment_method_ids() {
 		);
 	}
 
+	public function test_get_settings_request_returns_test_mode_flag() {
+		$this->gateway->update_option( 'test_mode', 'yes' );
+		add_filter(
+			'wcpay_dev_mode',
+			function () {
+				return true;
```

Comment:
Should dev mode be false in order to test both test_mode as yes|no?

Reply:
Sure thing - I made some changes in 9eb35c3c to test all the combinations